frontend/src/lib/Arrow.ts

```diff
@@ -0,0 +1,288 @@
+      if (isRangeIndex) {
+        const { start, stop, step } = field as RangeIndex
+        return range(start, stop, step).map(String)
+      }
+      const column = this.table.getColumn(field as string)
+      return Arrow.getColumnData(column).map(String)
+    })
+  }
+
+  private getColumns(): string[][] {
+    const isMultiIndex = this.schema.column_indexes.length > 1
+    return unzip(
+      this.schema.columns
+        .map(column => column.field_name)
+        .filter(fieldName => !this.schema.index_columns.includes(fieldName))
+        .map(fieldName =>
+          isMultiIndex
+            ? JSON.parse(
+                fieldName
+                  .replace(/\(/g, "[")
+                  .replace(/\)/g, "]")
+                  .replace(/'/g, '"')
+              )
+            : [fieldName]
+        )
+    )
+  }
+
+  private getData(): any[][] {
+    const rows = this.table.length
+    const columns = this.columns.length > 0 ? this.columns[0].length : 0
+
+    return range(0, rows).map(rowIndex =>
+      range(0, columns).map(columnIndex =>
+        this.table
+          .getColumnAt(columnIndex)
+          ?.get(rowIndex)
+          ?.toString()
+      )
+    )
+  }
+
+  get tableId(): string | undefined {
+    return this.styler?.uuid && `T_${this.styler.uuid}`
+  }
+
+  get tableStyles(): string | undefined {
+    return this.styler?.styles
+  }
+
+  get caption(): string | undefined {
+    return this.styler?.caption
+  }
+
+  get dimensions(): TableDimensions {
+    const [headerColumns, dataRowsCheck] = this.index.length
+      ? [this.index.length, this.index[0].length]
+      : [0, 0]
+
+    const [headerRows, dataColumnsCheck] = this.columns.length
+      ? [this.columns.length, this.columns[0].length]
+      : [0, 0]
+
+    const [dataRows, dataColumns] = this.data.length
+      ? [this.data.length, this.data[0].length]
+      : // If there is no data, default to the number of header columns.
+        [0, dataColumnsCheck]
+
+    if (
+      (dataRows !== 0 && dataRows !== dataRowsCheck) ||
+      (dataColumns !== 0 && dataColumns !== dataColumnsCheck)
+    ) {
+      throw new Error(
+        "Table dimensions don't align: " +
+          `rows(${dataRows} != ${dataRowsCheck}) OR ` +
+          `cols(${dataColumns} != ${dataColumnsCheck})`
+      )
+    }
+
+    const rows = headerRows + dataRows
+    const columns = headerColumns + dataColumns
+
+    return {
+      headerRows,
+      headerColumns,
+      dataRows,
+      dataColumns,
+      rows,
+      columns,
+    }
+  }
+
+  public getCell(rowIndex: number, columnIndex: number): TableCell {
+    const { headerRows, headerColumns, rows, columns } = this.dimensions
+
+    if (rowIndex < 0 || rowIndex >= rows) {
+      throw new Error("Row index is out of range.")
+    }
+    if (columnIndex < 0 || columnIndex >= columns) {
+      throw new Error("Column index is out of range.")
+    }
+
+    const isBlankCell = rowIndex < headerRows && columnIndex < headerColumns
```

Comment:
This sounds like a header cell instead but I'm not sure why this would be blank unless this is the cell where the row header and column header meets?